jwst/regtest/test_niriss.py

```diff
@@ -0,0 +1,67 @@
+"""Regression tests for NIRISS"""
+import os
+
+import pytest
+from astropy.io.fits.diff import FITSDiff
+
+from jwst.lib.suffix import replace_suffix
+from jwst.pipeline.collect_pipeline_cfgs import collect_pipeline_cfgs
+from jwst.stpipe import Step
+
+
+def is_like_truth(rtdata, fitsdiff_default_kwargs, suffix, truth_path='truth/niriss/test_niriss'):
+    """Compare step outputs with truth"""
+
+    # If the input was an association, the output should be the name of the product
+    # Otherwise, output is based on input.
+    if rtdata.asn:
+        output = rtdata.asn['products'][0]['name']
+    else:
+        output = os.path.splitext(os.path.basename(rtdata.input))[0]
+    output = replace_suffix(output, suffix) + '.fits'
+    rtdata.output = output
+
+    rtdata.get_truth(os.path.join(truth_path, output))
+
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+
+@pytest.fixture(scope='module')
+def run_nrs_wfss_spectral(jail, rtdata_module):
+    """Run the pipelines"""
+    rtdata = rtdata_module
+    rtdata.get_asn('niriss/level2a/nir_wfss_spec2_asn.json')
+
+    collect_pipeline_cfgs('config')
+    args = [
+        'config/calwebb_spec2.cfg',
+        rtdata.input,
+        '--steps.bkg_subtract.save_results=true',
```

Comment:
Not all of these steps get applied to WFSS data, such as imprint_subtract, msa_flagging, straylight, fringe, barshadow, and resample_spec, so at the very least it's not necessary to save all of these outputs and I'm also wondering if they get produced at all when a step is skipped? Does the pipeline only create the ones for steps that are actually run?

Reply:
I was going to edit out the ones that didn't produce anything. Will do that.

Reply:
Sounds good. In the meantime I've confirmed that steps that get skipped do NOT create an output, so leaving them in should be harmless.